common/common.go

```diff
@@ -433,15 +434,46 @@ func GetBlock(cache *BlockCache, height int) (*walletrpc.CompactBlock, error) {
 	return block, nil
 }
 
+// Filters out blocks that have spammy transactions
+func FilterSpammyBlock(block *walletrpc.CompactBlock, spamFilterThreshold int) *walletrpc.CompactBlock {
+	if spamFilterThreshold == 0 {
+		return block
+	}
+
+	// Make a copy of the block so we can modify it
+	newBlock := proto.Clone(block).(*walletrpc.CompactBlock)
+
+	// Filter out Sapling transactions above the threshold by removing epk and ciphertext
+	for _, tx := range newBlock.Vtx {
+		if len(tx.Outputs)+len(tx.Actions) > spamFilterThreshold {
+			// Remove epk and ciphertext for Sapling Outputs
+			for _, outputs := range tx.Outputs {
+				outputs.Ciphertext = nil
+				outputs.Epk = nil
+			}
+
+			// Remove epk and ciphertext and nullifier for Actions
+			for _, action := range tx.Actions {
+				action.Ciphertext = nil
+				action.EphemeralKey = nil
+				action.Nullifier = nil
```

Comment:
What are the security risks of not sending the nullifiers to the wallets? It doesn't seem to enable double-spend attacks since miners are still preventing that; could a double-spending transaction enter the mempool?

Reply:
I don't think it will cause any double-spending type issues, because `zcashd` already rejects any Txns that attempt to spend an already spent nullifier (I've tested this), so it never enters the mempool.